File: app/src/main/java/com/task/utils/ResourcesUtil.java
package com.task.utils;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.drawable.Drawable;

import com.task.App;

import static android.os.Build.VERSION.SDK_INT;
import static android.os.Build.VERSION_CODES.LOLLIPOP;

/**
 * Created by AhmedEltaher on 05/12/16.
 */

public class ResourcesUtil {
    private static Context context = App.getContext();
    private static Resources.Theme theme = App.getContext().getTheme();

    public static Drawable getDrawableById(int resId) {
        return SDK_INT >= LOLLIPOP ? context.getResources().getDrawable(resId, theme) :
            context.getResources().getDrawable(resId);
    }
    public static String getString(int resId) {
        return SDK_INT >= LOLLIPOP ? context.getResources().getString(resId) :
                context.getResources().getString(resId);
    }
}

File: app/src/main/java/com/task/utils/ResourcesUtil.kt
package com.task.utils

import android.graphics.drawable.Drawable
import android.os.Build
import com.task.App

/**
 * Created by AhmedEltaher on 05/12/16.
 */

class ResourcesUtil {
    companion object INSTANCE {
        private val context = App.context
        private val theme = App.context?.theme

        fun getDrawableById(resId: Int): Drawable {
            return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
                context!!.resources.getDrawable(resId, theme)
            else
                context!!.resources.getDrawable(resId)
        }

        fun getString(resId: Int): String {
            return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
                context!!.resources.getString(resId)
            else
                context!!.resources.getString(resId)
        }
    }
}
